analytics.py

```diff
@@ -0,0 +1,284 @@
+import math     #For all of the math used
+from .. import utils
+#------------------------------------------------------------------
+#
+# Joseph Cruz
+# analytics.py
+#
+# Stores the functions from point_pattern.py that perform analysis
+# on data.
+#------------------------------------------------------------------
+
+#++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++
+#
+# function mean_nearest_neighbors_randomPoints
+#
+# Determines the mean nearest neighbor of randomly generated points
+#++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++
+
+
+def mean_nearest_neighbors_randomPoints():
+    p = 99
+    n = 100
+    list_means = []
+    '''
+    print("Default p = 99 and n = 100\n")
+    input_change = input("Would you like to change the values? y/n")
+    if input_change == 'y' or input_change == 'Y':
+        p = input("Enter a value for p: ")
+        print("\n")
+        n = input("Enter a value for n: ")
+    '''
+    for i in range(p):
+        randPoints = utils.generate_rand_points(n) 
+        newMean = average_nearest_neighbor_distance(randPoints)
+        list_means.append(newMean)
+
+    return list_means
+
+#++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++
+#
+# function find_crit_points
+#
+# Finds the critical points (min and max) of a list
+#++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++
+
+
+def find_crit_points(list_means):
+    entries = list_means
+    maxEntry = 0
+    minEntry = 2
+    for i in range(len(list_means)):
+        if entries[i] > maxEntry:
+            maxEntry = entries[i]
+        if entries[i] < minEntry:
+            minEntry = entries[i]
+
+    return minEntry,maxEntry
```

Comment:
what about `min()` and `max()` or sorting and looking at the first and last values?